bin/check-azurerm-monitor-metric.rb

```diff
@@ -150,6 +157,20 @@ class CheckAzurermMonitorMetric < Sensu::Plugin::Check::CLI
          long: '--critical-under CRIT',
          proc: proc { |val| val.to_i }
 
+  option :base_url,
+         description: 'The Azure resource API URL.',
+         short: '-b URL',
+         long: '--base-url URL',
+         default: 'https://management.azure.com',
+         proc: proc { |val| val.chomp('/') }
+
+  option :lookback_period,
+         description:  'The amount of time (in seconds) from the current time to look back when retrieving the metric.  This should be long enough to capture the last value submitted.',
+         short: '-k PERIOD',
+         long: '--look-back-period PERIOD',
+         default: 600, # 10 min.  This should generally be enough time to capture the last value, without wasting API credits
+         proc: proc { |val| val.to_i }
+
   def run
     if config[:resource_id].to_s.empty? && config[:resource_name].to_s.empty?
       unknown 'resource id or resource name/group/type/namespece and subscription id must be provided'
```

Comment:
Given the above I think we can improve this, something along the lines of:
```
unknown "resource id, resource name/group/type/namespace, and subscription id are required. You passed: resource_id: #{config[:resource_id]}, resource_name: #{config[:resource_name]}"
```

Also the error message indicates that subscription id must be specified but we do not check that, we can just make it a required parameter. I think if we make resource id a parameter we should check if any of `$resource_( name|group|namespace)^` are valid rather than just checking 2 possible options. Again I am going off the old message so if resource_name is required we should just make it required but if one of any of those are required we can check that. Also if you want you can remove the default strings and check if they are `nil` instead. You can also do a `proc` to to the `.to_s` so you don't have to do that here as well every time you use it.

Reply:
The resource id is made up of resource name/group/type/namespace and subscription id (along an optional resource parent).  Therefore, if you specify the resource id, the rest isn't used.  I'll make changes to make this clearer.